pkg/controllers/provisioning/provisioner.go

```diff
@@ -371,25 +372,22 @@ func (p *Provisioner) Schedule(ctx context.Context) (scheduler.Results, error) {
 		).Info("found provisionable pod(s)")
 	}
 	// Mark in memory when these pods were marked as schedulable or when we made a decision on the pods
-	p.cluster.MarkPodSchedulingDecisions(results.PodErrors, pendingPods...)
-	p.MarkPodSchedulingDecisionsNodeRegistrationHealthy(ctx, results)
+	p.cluster.MarkPodSchedulingDecisions(ctx, results.PodErrors,
+		resources.MergePodsMaps(
```

Comment:
Can we put this in a package other than resources? This is also so specific to the results function, can this just be a method on results.

`results.NodePoolToPodMapping()`

Reply:
Done